go/pkg/filemetadata/filemetadata_test.go

```diff
@@ -66,10 +84,95 @@ func TestComputeDirectory(t *testing.T) {
 	}
 }
 
-func createFile(t *testing.T, executable bool, contents string) (string, error) {
+type testSymlinkCreationResult struct {
+	symlink string
+	target  string
+}
+
+func (sc *testSymlinkCreationResult) cleanup() {
```

Comment:
If everything's under temp, why do you need to clean it up?

Reply:
This is actually to follow https://github.com/bazelbuild/remote-apis-sdks/blob/f9b4b0d53c0ee1ad5b56a4721058471ce39e0d9f/go/pkg/filemetadata/filemetadata_test.go#L38

Reply:
Not sure I follow... Do you mean in the next patch? I meant to simplify as instead of cleaning up manually, create everything under a TempDir directory -- pass this directory to everything that creates files or symlinks if you need to reuse it.

Reply:
Sorry about the confusion. My point is that the existing file tests are doing this clean up manually (even if those filenames are temporary and unique), therefore I followed that for consistency.

Reply:
Hmm, right, now I wonder why I did that :-) But sure, let's keep it, then.

Reply:
Oops, i actually removed the `Clean()` method..